packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -151,8 +153,12 @@ export default function VisualEditor( { styles } ) {
 			wrapperBlockName: _wrapperBlockName,
 			wrapperUniqueId: getCurrentPostId(),
 			isBlockBasedTheme: editorSettings.__unstableIsBlockBasedTheme,
+			blockTypes: getBlockTypes(),
```

Comment:
For clarity and performance, limit how much data is selected:

```suggestion
			hasV3BlocksOnly: getBlockTypes().every( ( type ) => {
 				return type.apiVersion >= 3;
		 	} ),
```

Reply:
I don't think we can do this because you'd return a new array on every call?

Reply:
Ah no it's boolean. I don't remember why I did this then.

Reply:
Well, with `.every` you'd return a boolean, which is why it's nice. Right?

That said, realistically `getBlockTypes` rarely ever changes, so ignore my comment. 🙃  I guess an alternative take would be

```js
const hasV3BlocksOnly = useMemo( () => blockTypes.every( ... ), [ blockTypes ] )
```

but I doubt that it matters for a ~100 collection like block types.

Reply:
Fixed in https://github.com/WordPress/gutenberg/pull/48286/commits/8423f85ab99c83ac43a71b545ac64f68301812cf. :)